src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,92 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const initialFormFields: Movie = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  const [movie, setMovie] = useState(initialFormFields);
+
+  const validForm = movie.title.trim()
+    && movie.imgUrl.trim()
+    && movie.imdbUrl.trim()
+    && movie.imdbId.trim();
+
+  const increaseCount = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    onAdd(movie);
+    setCount(currentCount => currentCount + 1);
+    setMovie(initialFormFields);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={increaseCount}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={movie.title}
+        onChange={value => setMovie({
+          ...movie, title: value,
+        })}
```

Comment:
```suggestion
        onChange={(value, name) => setMovie({
          ...movie, [name]: value,
        })}
```
better to use event.target.name
for example like this (and pass to onChange in TextField component)
or even just pass reference to setMovie 
```suggestion
        onChange={setMovie}
```